File: Two Pointers/3Sum/ans.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        vector<vector<int>> ans;
        if(nums.size()<3){
            return {};
        }
        if(nums[0]>0){
            return {};
        }
        for(int i=0;i<nums.size();i++){
            if(nums[i]>0) break;
            if(i>0 && nums[i]==nums[i-1]) continue;
            int low=i+1;
            int high=nums.size()-1;
            int sum=0;
            while(low<high){
                sum=nums[i]+nums[low]+nums[high];
                if(sum<0)low++;
                else if(sum>0)high--;
                else{
                    ans.push_back({nums[i],nums[low],nums[high]});
                    int last_occurence =nums[low],last_high_occuremce=nums[high];
                    while(low<high && nums[low]==last_occurence){
                        low++;
                    }
                    while(low<high && nums[high]==last_high_occuremce){
                        high--;
                    }
                }
            }
        }
        return ans;
        
    }
};

File: Array&string/Roman to Interger/ans.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_values = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        sum = 0
        i = 0
        while i < len(s):
            if i + 1 < len(s) and roman_values[s[i]] < roman_values[s[i + 1]]:
                sum += roman_values[s[i + 1]] - roman_values[s[i]]
                i += 2
            else:
                sum += roman_values[s[i]]
                i += 1
        return sum